### PR TITLE
Write additonal tests and refactor existing code

### PR DESCRIPTION
**Pull request by kabirbaidhya**
* Write tests for `status.getCheckInterval()`.
 * Test `slack.isEnabled()` method.
 * Add negative test for `notifier.notify()` method.
 * Add `test:watch` npm script.
 * Stub slack.notify method on the notifier test.
 * Refactor notifier service to support services other than slack too.
 * Add nodejs version 5 & 4 in Travis CI config.
 * Test slack notifications are sent.
 * Add faker.
 * Use faker in the slack test.
 * Remove unnecessary files and dependencies.
 * Use async/await syntax.
 * Refactor monitoring logic using the `EventEmitter`.
 * Add `EVENT_MONITORING_STARTED` event.
 * Refactor monitoring logic using async/await.

This PR closes issue #5.

### CODE REVIEW

File: src/monitoring/eventListener.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import moment from 'moment';
+import logger from './../utils/logger';
+import { STATUS_UP } from '../services/status';
```
Comment (mesaugat) [inline src/monitoring/eventListener.js]: ↔️

File: src/monitoring/eventListener.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Handle status change event.
+ * 
+ * @param {any} params 
+ */
+function handleStatusChange(params) {
+  let { serviceName, status, time, lastStatusChanged } = params;
+  let notification = {
+    status,
+    name: serviceName,
+    time: params.time.clone(),
```
Comment (mesaugat) [inline src/monitoring/eventListener.js]: Why clone? So that it doesn't mutate?

Comment (kabirbaidhya) [inline src/monitoring/eventListener.js]: Yes, just to ensure things won't break when you mutate that date instance inside the other function.

File: src/services/events.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import EventEmitter from 'events';
+
+// Public constants
+export const EVENT_STATUS_CHANGED = 'status-changed';
+export const EVENT_MONITOING_STARTED = 'started-monitoring';
```
Comment (pratishshr) [inline src/services/events.js]: spell check: MONITORING*

Comment (kabirbaidhya) [inline src/services/events.js]: Good catch :+1:

File: src/services/notifier.js [hunk 4]
```diff
@@ -1,7 +1,24 @@
 import * as slack from './slack';
+import logger from './../utils/logger';
 
+/**
+ * List of notifier services.
+ */
+const notifiers = {
+  'slack': slack
```
Comment (pratishshr) [inline src/services/notifier.js]: any reason to not use the short-hand notation here?

Comment (kabirbaidhya) [inline src/services/notifier.js]: Can do that :+1:

**Comment by kabirbaidhya** [comment]
@leapfrogtechnology/the-chill-team PR has been updated, you guys can review now.

File: chill.test.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  maxInterval: 10000
+  method: OPTIONS
+  downStatus: '^(5..|4..)$'
+
+notifications:
+  slack:
+    enabled: false
+    endpoint: SLACK_WEBHOOK_ENDPOINT_HERE
+
+
```
Comment (mesaugat) [inline chill.test.yml]: Extra new line?

**Comment by mesaugat** [comment]
Now that we included `babel-polyfill`. Let's remove `babel-plugin-transform-runtime` from package.json and .babelrc

File: package.json [hunk 6]
```diff
@@ -17,11 +17,12 @@
     "lint:fix": "eslint src test --fix",
     "codecov": "cat coverage/*/lcov.info | codecov",
     "test": "NODE_ENV=test mocha",
+    "test:watch": "NODE_ENV=test mocha --watch",
     "test:coverage": "NODE_ENV=test istanbul cover node_modules/mocha/bin/_mocha -- --recursive --require babel-register --colors --reporter spec test"
```
Comment (mesaugat) [inline package.json]: I think we can use `babel-polyfill` here too and remove another dependency `babel-register`? Not sure.

Comment (kabirbaidhya) [inline package.json]: Not sure, it that works. We've used both here:
```
 --compilers js:babel-register
 --require babel-polyfill
 --recursive
```

Comment (mesaugat) [inline package.json]: Ok then. Let's leave it.

**Comment by mesaugat** [comment]
@leapfrogtechnology/the-chill-team We should merge this.

**Comment by pratishshr** [comment]
👍

**Comment by mesaugat** [comment]
Two ups, we are merging this right now.

**Pull request merged**